### PR TITLE
fix: prevent white space only text sending

### PR DESCRIPTION
**Pull request by AhyoungRyu**
### Description Of Changes
Resolves https://sendbird.atlassian.net/browse/UIKIT-3869

Previously, even I made the send btn invisible when there's only empty spaces(+ new lines) in the input field, but still the empty white spaces could be sent by hitting an enter button 😅

So I made a fix to block sending like below (see what I'm pressing)
recording

Plus I added more unit tests to prevent any further mistakes.

### CODE REVIEW

File: src/ui/MessageInput/index.jsx [hunk 1]
```diff
@@ -137,7 +137,7 @@ const MessageInput = React.forwardRef((props, ref) => {
       //
     }
     setMentionedUserIds([]);
-    setIsInput(textField?.innerText?.trim().length > 0);
+    setIsInput(textField?.textContent?.trim().length > 0);
```
Comment (AhyoungRyu) [inline src/ui/MessageInput/index.jsx:140]: Unfortunately jsdom doesn't recognize `innerText` So I had to replace it with `textContent` which is  similar to `innerText`  😕 
https://github.com/jsdom/jsdom/issues/1245

Otherwise we won't be able to add unit tests anymore.

Comment (sravan-s) [reply src/ui/MessageInput/index.jsx:140]: give me a little bit of time to test ~

File: src/ui/MessageInput/__tests__/MessageInput.spec.js [hunk 2]
```diff
@@ -45,28 +46,75 @@ describe('ui/MessageInput', () => {
...
-    const component = shallow(
-      <MessageInput
-        onSendMessage={noop}
-        message={{
-          message: 'hello',
-          mentionedMessageTempalte: 'hello'
-        }}
-      />
-    );
+  it('should call sendMessage with valid string', async () => {
```
Comment (sravan-s) [inline src/ui/MessageInput/__tests__/MessageInput.spec.js:49]: 👍

File: src/ui/MessageInput/__tests__/MessageInput.spec.js [hunk 3]
```diff
@@ -45,28 +46,75 @@ describe('ui/MessageInput', () => {
...
+
+    render(<MessageInput onSendMessage={onSendMessage} ref={textRef} />);
+
+    const input = screen.getByRole('textbox');
+    await userEvent.clear(input);
+    await userEvent.type(input, mockText);
+    expect(input.textContent).toBe(mockText);
+
+    fireEvent.keyDown(input, { key: 'Enter' });
+    expect(onSendMessage).toHaveBeenCalled();
```
Comment (sravan-s) [inline src/ui/MessageInput/__tests__/MessageInput.spec.js:62]: nit: Can we do a `toHaveBeenCalledWith` assertion instead?
 https://jestjs.io/docs/expect#tohavebeencalledwitharg1-arg2-

Comment (sravan-s) [reply src/ui/MessageInput/__tests__/MessageInput.spec.js:62]: Similar for other tests

File: src/ui/MessageInput/index.jsx (hunk 1 above)
Comment (sravan-s) [reply src/ui/MessageInput/index.jsx:140]: Need to test `onPaste` condition

Comment (AhyoungRyu) [reply src/ui/MessageInput/index.jsx:140]: How can I test the condition?

File: src/ui/MessageInput/__tests__/MessageInput.spec.js (hunk 3 above)
Comment (AhyoungRyu) [reply src/ui/MessageInput/__tests__/MessageInput.spec.js]: Yeah just did https://github.com/sendbird/sendbird-uikit-react/pull/538/commits/c128ae24d0b4531d61fb8e679e277ceb22e32260

File: src/ui/MessageInput/__tests__/MessageInput.spec.js [hunk 4]
```diff
@@ -45,28 +46,75 @@ describe('ui/MessageInput', () => {
     ).toBe(0);
   });
 
-  it.skip('should render send icon if text is present', () => {
```
Comment (AhyoungRyu) [inline src/ui/MessageInput/__tests__/MessageInput.spec.js:48]: Removed the skip tag from here and fixed the test inside here

File: src/ui/MessageInput/index.jsx (hunk 1 above)
Comment (sravan-s) [reply src/ui/MessageInput/index.jsx:140]: I wanted to do a smoke test manually haha
If you have free time, maybe add a test to detect XSS -> something like https://dev.to/lexlohr/unit-testing-with-jest-typescript-hands-on-session-345o#the-second-task

Comment (AhyoungRyu) [reply src/ui/MessageInput/index.jsx:140]: @sravan-s  So you're trying to test `onPaste` condition -> want me to add a test to detect XSS. I don't get what you really wanted to do 😅 Could you give me more detail what makes you think we need XSS test here?

File: src/ui/MessageInput/index.jsx [hunk 5]
```diff
@@ -420,6 +420,7 @@ const MessageInput = React.forwardRef((props, ref) => {
               e.preventDefault();
             } else {
               if (!e.shiftKey && e.key === MessageInputKeys.Enter
+                && textField?.textContent?.trim().length > 0
```
Comment (AhyoungRyu) [inline src/ui/MessageInput/index.jsx:423]: This newly added line is the only one directly related to "prevent white spaces only sending by hitting the enter key"

File: src/ui/MessageInput/index.jsx (hunk 1 above)
Comment (AhyoungRyu) [reply src/ui/MessageInput/index.jsx:140]: Had a short huddle with Sravan 🎧 
 - XSS test is not related to this change but kind of suggestion which is  good to have: probably will do separately
 - There could be some edge cases hardly caught by unit tests so manual tests is also needed

**Review by sravan-s — APPROVED** [review]

**Review by sravan-s — APPROVED** [review]
Tested manually ~ looks good

**Comment by AhyoungRyu** [comment]
@sravan-s Oh I manually tested from my side too but there's an issue when I try to mention someone's id. It throws an runtime error and it seemed related to this change. To fix this issue, I could modify other part of code but I thought it would be still risky. So I put the `innerText` back and only the part of that I newly added is using `textContent` which doesn't impact on the existing features. https://github.com/sendbird/sendbird-uikit-react/pull/538/commits/63c9fdd4208567da73ac10e0eb64962d9d652d3c 

Let a comment too.

File: src/ui/MessageInput/__tests__/MessageInput.spec.js [hunk 6]
```diff
@@ -92,7 +92,7 @@ describe('ui/MessageInput', () => {
     expect(onSendMessage).not.toHaveBeenCalledWith({ mentionTemplate: mockText, message: mockText });
   });
   
-  it('should render send icon if text is present', async() => {
```
Comment (AhyoungRyu) [inline src/ui/MessageInput/__tests__/MessageInput.spec.js]: Unfortunately had to put the skip tag again 🥲

File: src/ui/MessageInput/index.jsx (hunk 1 above)
Comment (AhyoungRyu) [reply src/ui/MessageInput/index.jsx:140]: Attaching this link too https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent#differences_from_innertext for someone who might wonder
